### PR TITLE
update checksum algorithms

### PR DESCRIPTION
**Pull request by Pnom**

### CODE REVIEW

**Review by DeadByElpy — APPROVED** [review]

File: tests/utils/lrc.test.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 
 import calculateLrc from '../../src/utils/calculateLrc.js';
+import getBytesFromHex from '../../src/utils/getBytesFromHex.js';
 
 
 describe('check lrc calculation', () => {
     test('check lrc calculation for simple sequence', () => {
-        const testSequence = new Uint8Array([0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x55]);
+        const testSequence = '00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f 55';
+        const testUin8Array = getBytesFromHex(testSequence);
```
Comment (DarkPark) [inline tests/utils/lrc.test.ts:10]:
```suggestion
        const testUint8Array = getBytesFromHex(testSequence);
```

or `testBytes`

File: tests/utils/crc32.test.ts [hunk 2]
```diff
@@ -1,11 +1,14 @@
-import {calculateCrc32, Crc32Type} from '../../src/utils/calculateCrc32.js';
+import {calculateCrc32} from '../../src/utils/calculateCrc32.js';
+import getBytesFromHex from '../../src/utils/getBytesFromHex.js';
 
 
 describe('check crc32 calculation', () => {
     test('check crc32 IEEE_8023 calculation', () => {
```
Comment (DarkPark) [inline tests/utils/crc32.test.ts:6]:
```suggestion
    test('IEEE_8023', () => {
```
?

and in other tests

File: tests/utils/crc16.test.ts [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
 describe('check crc16 calculation', () => {
     test('check crc16 x25 calculation', () => {
-        const testSequence = [0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f];
+        const testSequence = '00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f';
+        const testUin8Array = getBytesFromHex(testSequence);
+        const testArray = Array.from(testUin8Array);
 
-        expect(calculateCrc16(Crc16Type.X25, testSequence)).toBe(0x13E9);
-        expect(calculateCrc16(Crc16Type.X25, new Uint8Array(testSequence))).toBe(0x13E9);
+        expect(calculateCrc16(testUin8Array)).toBe(0x13E9);
```
Comment (DarkPark) [inline tests/utils/crc16.test.ts:11]: usually we use lower case for hex

File: src/utils/calculateCrc32.ts [hunk 4]
```diff
@@ -31,12 +31,11 @@ export enum Crc32Type {
  *
  * @return CRC32
  */
+export const calculateCrc32 = ( data: Uint8Array | Array<number>, crc32type: Crc32Type = Crc32Type.IEEE_8023 ) => {
```
Comment (DarkPark) [inline src/utils/calculateCrc32.ts:34]: export default ?

File: src/utils/calculateCrc16.ts [hunk 5]
```diff
@@ -25,9 +27,12 @@ export enum Crc16Type {
  *
  * @return CRC32
  */
+export const calculateCrc16 = ( data: Uint8Array | Array<number>, crc16type: Crc16Type = Crc16Type.X25 ) => {
```
Comment (DarkPark) [inline src/utils/calculateCrc16.ts:30]: export default ?

File: tests/utils/lrc.test.ts (hunk 1 above)
Comment (Pnom) [reply tests/utils/lrc.test.ts:10]: fixed

File: tests/utils/crc32.test.ts (hunk 2 above)
Comment (Pnom) [reply tests/utils/crc32.test.ts]: fixed

File: tests/utils/crc16.test.ts (hunk 3 above)
Comment (Pnom) [reply tests/utils/crc16.test.ts]: fixed

File: src/utils/calculateCrc16.ts (hunk 5 above)
Comment (Pnom) [reply src/utils/calculateCrc16.ts]: fixed

File: tests/utils/lrc.test.ts [hunk 6]
```diff
@@ -1,12 +1,16 @@
 /* eslint-disable object-curly-newline */
 
 import calculateLrc from '../../src/utils/calculateLrc.js';
+import getBytesFromHex from '../../src/utils/getBytesFromHex.js';
 
 
 describe('check lrc calculation', () => {
     test('check lrc calculation for simple sequence', () => {
```
Comment (DarkPark) [inline tests/utils/lrc.test.ts:8]:
```suggestion
    test('simple sequence', () => {
```
?

Comment (Pnom) [reply tests/utils/lrc.test.ts]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**